subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/OutputScrapingExecutionResult.java

```diff
@@ -220,10 +224,39 @@ public ExecutionResult assertNotOutput(String expectedOutput) {
         return this;
     }
 
+    private static class LineWithDistance {
+        private final String line;
+        private final int distance;
+
+        public LineWithDistance(String line, int distance) {
+            this.line = line;
+            this.distance = distance;
+        }
+
+        public String getLine() {
+            return line;
+        }
+
+        public int getDistance() {
+            return distance;
+        }
+    }
+
     @Override
     public ExecutionResult assertContentContains(String actualText, String expectedOutput, String label) {
         String expectedText = LogContent.of(expectedOutput).withNormalizedEol();
         if (!actualText.contains(expectedText)) {
+            if (!expectedText.contains("\n")) {
+                Arrays.stream(actualText.split("\n"))
+                    // Measure Levenshtein distance for each line
+                    .map(line -> new LineWithDistance(line, StringUtils.getLevenshteinDistance(line, expectedText)))
```

Comment:
Running Levensthein distance over each line of output text - does this get noticeably slower in wall-clock time when the test has more than a few output lines?  I think this is `O(actual line length * expected line length * num actual lines)`, so it shouldn't, but maybe worth a test against some more substantial output?  Maybe this stream could also be `parallel()`-ized?